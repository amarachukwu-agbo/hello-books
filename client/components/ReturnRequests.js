import React from 'react';

const ReturnRequests = (props) => {
  const { books } = props;
  if (!books.length) {
    return (
        <div className="row center">
            <p className="grey-text">You have no pending return requests </p>
        </div>
    );
  }

  return (
        <div className="row">
            <table className="striped responsive-table">
                <thead>
                    <tr>
                        <th>Book</th>
                        <th>Author</th>
                        <th>Date</th>
                        <th>Status</th>
                    </tr>
                </thead>
                <tbody>
                    { books.map((book, index) =>
                        <tr key={ index }>
                            <td> { book.returnRequests.title } </td>
                            <td> { book.returnRequests.author } </td>
                            <td> { book.createdAt.split('T')[0] } </td>
                            <td> <button className="btn-flat btn-small">{ book.status } </button></td>
                        </tr>)}
                </tbody>
            </table>
        </div>
  );
};

export default ReturnRequests;
