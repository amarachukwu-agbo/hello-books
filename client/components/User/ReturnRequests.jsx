import React from 'react';
import propTypes from 'prop-types';

/**
 * @description stateless component for rendering a user's
 * return requests
 *
 * @param {object} props - properties of the component
 *
 * @returns {Node} - react node containing the ReturnRequests component
 */

const ReturnRequests = (props) => {
  const { requests } = props;
  if (!requests.length) {
    return (
      <div className="row center">
        <p className="grey-text">You have no pending return requests </p>
      </div>
    );
  }

  return (
    <div className="section">
      <table className="striped responsive-table">
        <thead>
          <tr>
            <th>Book</th>
            <th>Author</th>
            <th>Date</th>
            <th>Status</th>
          </tr>
        </thead>
        <tbody>
          { requests.map((request, index) =>
            <tr key={index}>
              <td> {request.returnRequests.title} </td>
              <td> {request.returnRequests.author} </td>
              <td> {request.createdAt.split('T')[0]} </td>
              <td>
                <button className="btn-flat btn-small">
                  <span className={request.status === 'Pending' ?
                  'orange-text' : 'red-text'}>
                    &#9679; </span>
                  {request.status}
                </button>
              </td>
            </tr>)}
        </tbody>
      </table>
    </div>
  );
};

// Prop types validation
ReturnRequests.propTypes = {
  requests: propTypes.array.isRequired,
};
export default ReturnRequests;
