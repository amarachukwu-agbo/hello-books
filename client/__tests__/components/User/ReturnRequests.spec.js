import React from 'react';
import { shallow } from 'enzyme';
import mockData from '../../__mocks__/mockData';
import ReturnRequests from '../../../components/User/ReturnRequests';

const { users } = mockData;
const props = {
  requests: users.profileResponse.user.userReturnRequests,
};
describe('<ReturnRequests />', () => {
  it('should render return requests correctly', () => {
    const tree = shallow(<ReturnRequests {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('should render <p> if there are no return requests', () => {
    const requests = [];
    const wrapper = shallow(<ReturnRequests requests = { requests } />);
    expect(wrapper.find('p').text())
      .toEqual('You have no pending return requests');
  });
});
