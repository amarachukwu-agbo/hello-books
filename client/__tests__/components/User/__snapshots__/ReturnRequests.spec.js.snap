// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReturnRequests /> should render return requests correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReturnRequests
    requests={
      Array [
        Object {
          "bookId": 99,
          "comments": null,
          "createdAt": "2018-06-24T01:14:32.115Z",
          "id": 4,
          "returnRequests": Object {
            "author": "Andrew Hart",
            "title": "Lies That Bind Us",
          },
          "status": "Accepted",
          "updatedAt": "2018-06-24T01:20:19.244Z",
          "userId": 3,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <table
        className="striped responsive-table"
      >
        <thead>
          <tr>
            <th>
              Book
            </th>
            <th>
              Author
            </th>
            <th>
              Date
            </th>
            <th>
              Status
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
               
              Lies That Bind Us
               
            </td>
            <td>
               
              Andrew Hart
               
            </td>
            <td>
               
              2018-06-24
               
            </td>
            <td>
              <button
                className="btn-flat btn-small"
              >
                <span
                  className="red-text"
                >
                  ● 
                </span>
                Accepted
              </button>
            </td>
          </tr>
        </tbody>
      </table>,
      "className": "section",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <thead>
            <tr>
              <th>
                Book
              </th>
              <th>
                Author
              </th>
              <th>
                Date
              </th>
              <th>
                Status
              </th>
            </tr>
          </thead>,
          <tbody>
            <tr>
              <td>
                 
                Lies That Bind Us
                 
              </td>
              <td>
                 
                Andrew Hart
                 
              </td>
              <td>
                 
                2018-06-24
                 
              </td>
              <td>
                <button
                  className="btn-flat btn-small"
                >
                  <span
                    className="red-text"
                  >
                    ● 
                  </span>
                  Accepted
                </button>
              </td>
            </tr>
          </tbody>,
        ],
        "className": "striped responsive-table",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr>
              <th>
                Book
              </th>
              <th>
                Author
              </th>
              <th>
                Date
              </th>
              <th>
                Status
              </th>
            </tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th>
                  Book
                </th>,
                <th>
                  Author
                </th>,
                <th>
                  Date
                </th>,
                <th>
                  Status
                </th>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Book",
                },
                "ref": null,
                "rendered": "Book",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Author",
                },
                "ref": null,
                "rendered": "Author",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Date",
                },
                "ref": null,
                "rendered": "Date",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Status",
                },
                "ref": null,
                "rendered": "Status",
                "type": "th",
              },
            ],
            "type": "tr",
          },
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <tr>
                <td>
                   
                  Lies That Bind Us
                   
                </td>
                <td>
                   
                  Andrew Hart
                   
                </td>
                <td>
                   
                  2018-06-24
                   
                </td>
                <td>
                  <button
                    className="btn-flat btn-small"
                  >
                    <span
                      className="red-text"
                    >
                      ● 
                    </span>
                    Accepted
                  </button>
                </td>
              </tr>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "4",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <td>
                     
                    Lies That Bind Us
                     
                  </td>,
                  <td>
                     
                    Andrew Hart
                     
                  </td>,
                  <td>
                     
                    2018-06-24
                     
                  </td>,
                  <td>
                    <button
                      className="btn-flat btn-small"
                    >
                      <span
                        className="red-text"
                      >
                        ● 
                      </span>
                      Accepted
                    </button>
                  </td>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "Lies That Bind Us",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "Lies That Bind Us",
                    " ",
                  ],
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "Andrew Hart",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "Andrew Hart",
                    " ",
                  ],
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "2018-06-24",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "2018-06-24",
                    " ",
                  ],
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <button
                      className="btn-flat btn-small"
                    >
                      <span
                        className="red-text"
                      >
                        ● 
                      </span>
                      Accepted
                    </button>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <span
                          className="red-text"
                        >
                          ● 
                        </span>,
                        "Accepted",
                      ],
                      "className": "btn-flat btn-small",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "● ",
                          "className": "red-text",
                        },
                        "ref": null,
                        "rendered": "● ",
                        "type": "span",
                      },
                      "Accepted",
                    ],
                    "type": "button",
                  },
                  "type": "td",
                },
              ],
              "type": "tr",
            },
          ],
          "type": "tbody",
        },
      ],
      "type": "table",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <table
          className="striped responsive-table"
        >
          <thead>
            <tr>
              <th>
                Book
              </th>
              <th>
                Author
              </th>
              <th>
                Date
              </th>
              <th>
                Status
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                 
                Lies That Bind Us
                 
              </td>
              <td>
                 
                Andrew Hart
                 
              </td>
              <td>
                 
                2018-06-24
                 
              </td>
              <td>
                <button
                  className="btn-flat btn-small"
                >
                  <span
                    className="red-text"
                  >
                    ● 
                  </span>
                  Accepted
                </button>
              </td>
            </tr>
          </tbody>
        </table>,
        "className": "section",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead>
              <tr>
                <th>
                  Book
                </th>
                <th>
                  Author
                </th>
                <th>
                  Date
                </th>
                <th>
                  Status
                </th>
              </tr>
            </thead>,
            <tbody>
              <tr>
                <td>
                   
                  Lies That Bind Us
                   
                </td>
                <td>
                   
                  Andrew Hart
                   
                </td>
                <td>
                   
                  2018-06-24
                   
                </td>
                <td>
                  <button
                    className="btn-flat btn-small"
                  >
                    <span
                      className="red-text"
                    >
                      ● 
                    </span>
                    Accepted
                  </button>
                </td>
              </tr>
            </tbody>,
          ],
          "className": "striped responsive-table",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th>
                  Book
                </th>
                <th>
                  Author
                </th>
                <th>
                  Date
                </th>
                <th>
                  Status
                </th>
              </tr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th>
                    Book
                  </th>,
                  <th>
                    Author
                  </th>,
                  <th>
                    Date
                  </th>,
                  <th>
                    Status
                  </th>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Book",
                  },
                  "ref": null,
                  "rendered": "Book",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Author",
                  },
                  "ref": null,
                  "rendered": "Author",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Date",
                  },
                  "ref": null,
                  "rendered": "Date",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Status",
                  },
                  "ref": null,
                  "rendered": "Status",
                  "type": "th",
                },
              ],
              "type": "tr",
            },
            "type": "thead",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <tr>
                  <td>
                     
                    Lies That Bind Us
                     
                  </td>
                  <td>
                     
                    Andrew Hart
                     
                  </td>
                  <td>
                     
                    2018-06-24
                     
                  </td>
                  <td>
                    <button
                      className="btn-flat btn-small"
                    >
                      <span
                        className="red-text"
                      >
                        ● 
                      </span>
                      Accepted
                    </button>
                  </td>
                </tr>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "4",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <td>
                       
                      Lies That Bind Us
                       
                    </td>,
                    <td>
                       
                      Andrew Hart
                       
                    </td>,
                    <td>
                       
                      2018-06-24
                       
                    </td>,
                    <td>
                      <button
                        className="btn-flat btn-small"
                      >
                        <span
                          className="red-text"
                        >
                          ● 
                        </span>
                        Accepted
                      </button>
                    </td>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        "Lies That Bind Us",
                        " ",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      "Lies That Bind Us",
                      " ",
                    ],
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        "Andrew Hart",
                        " ",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      "Andrew Hart",
                      " ",
                    ],
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        "2018-06-24",
                        " ",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      "2018-06-24",
                      " ",
                    ],
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <button
                        className="btn-flat btn-small"
                      >
                        <span
                          className="red-text"
                        >
                          ● 
                        </span>
                        Accepted
                      </button>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <span
                            className="red-text"
                          >
                            ● 
                          </span>,
                          "Accepted",
                        ],
                        "className": "btn-flat btn-small",
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "● ",
                            "className": "red-text",
                          },
                          "ref": null,
                          "rendered": "● ",
                          "type": "span",
                        },
                        "Accepted",
                      ],
                      "type": "button",
                    },
                    "type": "td",
                  },
                ],
                "type": "tr",
              },
            ],
            "type": "tbody",
          },
        ],
        "type": "table",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
